src/library_glfw.js

```diff
@@ -713,38 +713,50 @@ var LibraryGLFW = {
 
     setKeyCallback: function(winid, cbfun) {
       var win = GLFW.WindowFromId(winid);
-      if (!win) return;
+      if (!win) return null;
```

Comment:
Hmm, was there a specific reason to `return null;` instead of just `return;`? That should be functionally the same (returns `undefined` but no use differentiating between `undefined` and `null`), and is shorter code size -wise?

Reply:
Otherwise, looks good!

Reply:
The null is the same of the default value for callback functions. We should implement behavior that described in the `GLFW` documentation.
We also implement `C` API where function could not return value of a different types.

Reply:
Oh, JS `null` is different from C `null`. If a JS function returns any of `undefined`, `null` or `0` back to compiled C code, it will be a C `null`/`0` in there, so a `return;` would do the same. Though it is more of a micro-optimization on code size, so perhaps doesn't matter that much.

Reply:
Thanks for the description.  I started as an Emscripten user, but now I'm trying to fix bugs.

Reply:
There is some risk of the code running more slowly if the JS engine needs to change types. So it is probably best in these cases to `return 0` since `0` is the type it will be coerced into anyhow to reach compiled code.

However, these code paths are probably not super-hot. So for code size no return at all may be best. It does make the code less clear to read, though. Maybe an option is to emit something the optimizer can later remove.